.github/workflows/ci.yml

```diff
@@ -228,15 +228,58 @@ jobs:
           cd c_examples
           make
 
-  publish:
+  release:
+    # create a Github release
+    # only when
+    # - push on master
+    # - tag starts with 'v*'
     needs: [format, dummy, xen, kvm, virtualbox, hyper-v, c_api]
     runs-on: ubuntu-20.04
+    # output these value to be used by other jobs so they can add assets
+    outputs:
+      upload_url: ${{ steps.step_upload_url.outputs.upload_url }}
+      version: ${{ steps.get_version.outputs.version }}
+
+    if: github.event_name == 'push' && startsWith(github.ref, 'refs/tags/v')
+    steps:
+      - uses: actions/checkout@v1
+
+      - name: Get the version
+        id: get_version
+        run: echo ::set-output name=VERSION::${GITHUB_REF/refs\/tags\//}
+
+      - name: install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+
+      - name: install cargo changelog
```

Comment:
This is step could have been joined with the step above.

Reply:
so any subcommand is a valid toolchain component ?
I wanted to do that at the beginning, then I went to check what a toolchain component was, and ended up on this
https://rust-lang.github.io/rustup/concepts/components.html

As it was not listed, I figured it did not qualify as a component.

Also, I just tried:
~~~
$ rustup component add changelog
error: toolchain 'stable-x86_64-unknown-linux-gnu' does not contain component 'changelog' for target 'x86_64-unknown-linux-gnu'
~~~

Reply:
Sorry, my bad. I was probably thinking of this: https://github.com/Wenzel/libmicrovmi/blob/31f52ba297e07aac3828514382fca2755c8d77c8/.github/workflows/security-audit.yml#L20